dataclasses_json/core.py

```diff
@@ -140,6 +140,9 @@ def _is_supported_generic(type_):
 def _decode_generic(type_, value, infer_missing):
     if value is None:
         res = value
+    elif _issubclass_safe(type_, Enum):
+        # Convert to an Enum using the type as a constructor. Assumes a direct match is found.
+        res = type_(value)
```

Comment:
@dpausp  should we also consider adding a fallback to `type_[value]` for enum here to be able to parse the enum by name if the parsing by value `type_(value)` fails?